src/data/matrix/basic.lean

```diff
@@ -1441,6 +1441,12 @@ lemma conj_transpose_reindex [has_star α] (eₘ : m ≃ l) (eₙ : n ≃ o) (M
   (reindex eₘ eₙ M)ᴴ = (reindex eₙ eₘ Mᴴ) :=
 rfl
 
+lemma mul_transpose_eq_reindex_mul_reindex_transpose [fintype n] [fintype m] [semiring α]
+  (e : m ≃ n) (M : matrix n m α) :
+  M ⬝ Mᵀ = (reindex (equiv.refl n) e M) ⬝ (reindex (equiv.refl n) e M)ᵀ :=
```

Comment:
Is there a reason not to state this as:
```suggestion
  M ⬝ Mᵀ = (M.minor id e) ⬝ (M.minor id e)ᵀ :=
```
instead? `minor` is the simp-normal form.